src/Draco.Compiler/Internal/Documentation/Extractors/XmlDocumentationExtractor.cs

```diff
@@ -0,0 +1,112 @@
+    public static SymbolDocumentation Extract(Symbol containingSymbol) =>
+        new XmlDocumentationExtractor(containingSymbol.RawDocumentation, containingSymbol).Extract();
+
+    /// <summary>
+    /// Extracts the <see cref="Xml"/>.
+    /// </summary>
+    /// <returns>The extracted XMl as <see cref="SymbolDocumentation"/>.</returns>
+    private SymbolDocumentation Extract()
+    {
+        // TODO: exception
+        //       para
+        //       list
+        //       c
+        //       see - not cref links
+        //       seealso
+        //       b ?
+        //       i ?
+
+        var xml = $"""
+            <documentation>
+              {this.Xml}
+            </documentation>
+            """;
+        var doc = new XmlDocument();
+        doc.LoadXml(xml);
+
+        var raw = doc.DocumentElement!.ChildNodes
+            .Cast<XmlNode>()
+            .Select(this.ExtractSectionOrElement);
+
+        var sections = raw.OfType<DocumentationSection>().ToList();
+        var elements = raw.OfType<DocumentationElement>();
+
+        foreach (var grouped in elements.GroupBy(x => x.GetType()))
+        {
+            if (grouped.Key == typeof(ParameterDocumentationElement)) sections.Add(new DocumentationSection(SectionKind.Parameters, grouped.ToImmutableArray()));
+            else if (grouped.Key == typeof(TypeParameterDocumentationElement)) sections.Add(new DocumentationSection(SectionKind.TypeParameters, grouped.ToImmutableArray()));
+        }
+        return new SymbolDocumentation(sections.ToImmutableArray());
+    }
+
+    private object ExtractSectionOrElement(XmlNode node) => node.Name switch
+    {
+        "param" => new ParameterDocumentationElement(this.GetParameter(node.Attributes?["name"]?.Value ?? string.Empty), this.ExtractElementsFromNode(node)),
+        "typeparam" => new TypeParameterDocumentationElement(this.GetTypeParameter(node.Attributes?["name"]?.Value ?? string.Empty), this.ExtractElementsFromNode(node)),
+        "code" => new DocumentationSection(SectionKind.Code, ImmutableArray.Create(this.ExtractElement(node))),
+        "summary" => new DocumentationSection(SectionKind.Summary, this.ExtractElementsFromNode(node)),
+        _ => new DocumentationSection(node.Name, this.ExtractElementsFromNode(node)),
+    };
+
+    private DocumentationElement ExtractElement(XmlNode node) => node.LocalName switch
+    {
+        "#text" => new TextDocumentationElement(node.InnerText),
+        "see" => string.IsNullOrEmpty(node.InnerText)
+            ? new ReferenceDocumentationElement(this.GetSymbolFromDocumentationName(node.Attributes?["cref"]?.Value ?? string.Empty) ?? new PrimitiveTypeSymbol(node.Attributes?["cref"]?.Value[2..] ?? string.Empty, false))
+            : new ReferenceDocumentationElement(this.GetSymbolFromDocumentationName(node.Attributes?["cref"]?.Value ?? string.Empty) ?? new PrimitiveTypeSymbol(node.Attributes?["cref"]?.Value[2..] ?? string.Empty, false), node.InnerText),
```

Comment:
Can't the conditional be moved in for the last param to be null or node.InnerText?